wandb/wandb_run.py

```diff
@@ -418,18 +418,17 @@ def use_artifact(self, artifact, type=None, aliases=None):
         if self.mode == "dryrun":
             wandb.termwarn("Using artifacts in dryrun mode is currently unsupported.")
             return artifact
-        if type is not None:
-            wandb.termwarn("Type argument is deprecated and will be removed in a future release.")
         self.history.ensure_jupyter_started()
         if isinstance(artifact, str):
-            if type is None:
-                raise ValueError('type required')
             # Ensure we have an entity (_load_entity puts the result into api.settings())
             # then initialize PublicApi with it.
             self._load_entity(self.api, True)
             public_api = PublicApi(self.api.settings())
             artifact = public_api.artifact(name=artifact)
             self.api.use_artifact(artifact.id)
+            if type is not None and type != artifact.type:
```

Comment:
Maybe we do this before calling `use_artifact`?

Reply:
Good call, we definitely should to avoid making the ref.